client/skills.go

```diff
@@ -0,0 +1,149 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// Skill represents a capacity offered by a snap.
+type Skill struct {
+	Name  string                 `json:"name"`
+	Snap  string                 `json:"snap"`
+	Type  string                 `json:"type,omitempty"`
+	Attrs map[string]interface{} `json:"attrs,omitempty"`
+	Apps  []string               `json:"apps,omitempty"`
+	Label string                 `json:"label,omitempty"`
+}
+
+// Slot represents the potential of a given snap to use a skill.
+type Slot struct {
+	Name  string                 `json:"name"`
+	Snap  string                 `json:"snap"`
+	Type  string                 `json:"type,omitempty"`
+	Attrs map[string]interface{} `json:"attrs,omitempty"`
+	Apps  []string               `json:"apps,omitempty"`
+	Label string                 `json:"label,omitempty"`
+}
+
+// SkillUsage represents the usage of a single skill.
+type SkillUsage struct {
```

Comment:
This doesn't seem quite right on a first look. The entities are the skill and the slot.. whose name, snap, etc, are these referring to, and why do we not have the actual thing here instead of picking up individual fields?

Needs something further consideration..

I'd also avoid the term "usage".. note that the term "uses" creates a slot in a snap. The commands this data is talking about are "grant" and "revoke". That's what people will be looking for.